climada/util/api_client.py

```diff
@@ -486,3 +493,145 @@ def purge_cache(self, local_path):
+                    break
+                else:
+                    LOGGER.error('Please give a valid value from the list provided.')
+        datasets = self._get_data(exposures_type)
+        exposures_list = []
+        for dataset in datasets:
+            self.download_file(data_dir, dataset.files[0])
+            if os.path.isfile(os.path.join(data_dir, dataset.files[0].file_name)):
+                LOGGER.info('The file already exists and it was not downloaded again.')
+            exposures = Exposures()
+            exposures.read_hdf5(os.path.join(data_dir, dataset.files[0].file_name))
+            exposures_list.append(exposures)
+        exposures_concat = Exposures()
+        exposures_concat = exposures_concat.concat(exposures_list)
+        return exposures_concat
+
+    def get_litpot_default(self, country=None, data_dir=SYSTEM_DIR):
+        """Get a LitPop instance on a 150arcsec grid with the default parameters: exponents:(1,1) and fin_mode='pc'.
+          Parameters
+          ----------
+          country : list
+              List of country for which to create the LitPop object. If None is given, a global LitPop instance
+              is created
+          data_dir : str
+              directory where the files should be downoladed. Default: SYSTEM_DIR
+          """
+        if not country:
+            datasets = [self.get_datasets(data_type='litpop', properties={'exponents': '(1,1)', 'fin_mode': 'pc'})]
+        if country:
+            country = list(country)
+            datasets = [self.get_datasets(data_type='litpop', properties={'exponents': '(1,1)', 'fin_mode': 'pc',
+                                                                             'country_iso3alpha': c}) for c in country]
+        exposures_list = []
+        for dataset in datasets:
+            if os.path.isfile(os.path.join(data_dir, dataset.files[0])):
+                LOGGER.info('The file already exists and it was not downloaded again.')
+            self.download_file(data_dir, dataset.files[0])
+            exposures = Exposures()
+            exposures.read_hdf5(os.path.join(data_dir, dataset.files[0].file_name))
+            exposures_list.append(exposures)
+        exposures_concat = Exposures()
+        exposures_concat = exposures_concat.concat(exposures_list)
+        return exposures_concat
+
+    def _get_data(self, type):
+        datasets = self.get_datasets(data_type=type)
+        properties_keys = np.unique([dataset.properties.keys() for dataset in datasets])
+        # find common properties between "groups" of datasets:
+        properties_keys = set(properties_keys[0]).intersection(*properties_keys)
+        user_properties_input = {}
+        # get user input to differentiate between these groups
+        user_properties_input = self._select_properties(datasets, properties_keys, user_properties_input)
+        datasets = self.get_datasets(data_type=type, properties=user_properties_input)
```

Comment:
any idea how to not query the datasets a second time here with the extra properties?